File: CheckImageView/app/src/main/java/deadline/checkImageView/ICheckAttacher.java
package deadline.checkImageView;

/**
 * @author deadline
 */

public interface ICheckAttacher {

    float DEFAULT_CHECKED_SCALE = 1.2f;
    float DEFAULT_UNCHECKED_SCALE = 1.0f;
    long ZOOM_DURATION = 300L;


    interface OnCheckedChangeListener{

        void onChange(boolean checked);
    }


    boolean isChecked();

    void setChecked(boolean checked);

    void setCheckedScale(float scale);

    float getScale();

    void setZoomAnimationDuration(long duration);

    void setOnCheckChangeListener(OnCheckedChangeListener listener);
}

File: CheckImageView/app/src/main/java/kot/checkImageView/ICheckAttacher.kt
package kot.checkImageView

/**
 * @author deadline
 * @time 2017/10/13
 */

val DEFAULT_CHECKED_SCALE = 1.2f
val DEFAULT_UNCHECKED_SCALE = 1.0f
val ZOOM_DURATION = 300L

interface ICheckAttacher {

    fun isChecked(): Boolean

    fun setChecked(checked: Boolean)

    fun setCheckedScale(scale: Float)

    fun getScale(): Float

    fun setZoomAnimationDuration(duration: Long)

    fun setOnCheckChangeListener(listener: OnCheckedChangeListener)
}

interface OnCheckedChangeListener{

    fun onChange(checked: Boolean)
}